specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json

```diff
@@ -0,0 +1,1120 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Action Groups API",
+    "x-ms-code-generation-settings": {
+      "name": "MonitorManagementClient"
+    },
+    "version": "2021-09-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/actionGroups/{actionGroupName}": {
+      "put": {
+        "description": "Create a new action group or update an existing one.",
+        "operationId": "ActionGroups_CreateOrUpdate",
+        "x-ms-examples": {
+          "Create or update an action group": {
+            "$ref": "./examples/createOrUpdateActionGroup.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ActionGroupNameParameter"
+          },
+          {
+            "name": "actionGroup",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ActionGroupResource"
+            },
+            "description": "The action group to create or use for the update."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "An existing action group was successfully updated.",
+            "schema": {
+              "$ref": "#/definitions/ActionGroupResource"
+            }
+          },
+          "201": {
+            "description": "A new action group was successfully created.",
+            "schema": {
+              "$ref": "#/definitions/ActionGroupResource"
+            }
+          },
+          "default": {
+            "description": "An error occurred and the action group could not be created or updated.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
```

Comment:
Could refactor to reference ErrorResponse from https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json